vm-virtio/src/device.rs

```diff
@@ -101,3 +101,15 @@ pub trait VirtioDevice: Send {
 pub trait DmaRemapping: Send + Sync {
     fn translate(&self, id: u32, addr: u64) -> std::result::Result<u64, std::io::Error>;
 }
+
+/// Trait meant for triggering the DMA mapping update related to an external
+/// device not managed fully through virtio. It is dedicated to virtio-iommu
+/// in order to trigger the map update anytime the mapping is updated from the
+/// guest.
+pub trait ExternalDmaMapping: Send + Sync {
+    /// Map a memory range
+    fn map(&self, iova: u64, gpa: u64, size: u64) -> std::result::Result<(), std::io::Error>;
+
+    /// Unmap a memory range
+    fn unmap(&self, iova: u64, size: u64) -> std::result::Result<(), std::io::Error>;
+}
```

Comment:
Well we cannot have the trait declared by a consumer of vm-virtio, otherwise we would end up with some cyclic dependency, right?
We could define an independent crate for this kind of thing, since both vfio and vm-virtio would need to rely on it.

Reply:
Are we sure the `vm-device` will be completely `vm-virtio` and `vfio` free? This is important as we would run into a cyclic dependency later if this was happening.
Also, do you want me to create the vm-device crate in our repo, or should we do this later?